x/shared/helpers/service_test.go

```diff
@@ -28,6 +32,78 @@ func TestIsValidServiceId(t *testing.T) {
 	}
 }
 
+func TestIsValidServiceName(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected bool
+	}{
+		{"ValidName-1", true},
+		{"Valid Name_1", true},
+		{"valid name with spaces", true},
+		{"invalid@name", false}, // contains invalid character '@'
+		{"Valid.Name", false},   // contains invalid character '.'
+		{"", true},              // empty string is valid for ServiceName
+		{"validnamebuttoolongvalidnamebuttoolongvalidnamebuttoolong", false}, // exceeds maxServiceIdName length
+	}
+
+	for _, test := range tests {
+		t.Run(test.input, func(t *testing.T) {
+			result := IsValidServiceName(test.input)
+			if result != test.expected {
+				t.Errorf("For input %s, expected %v but got %v", test.input, test.expected, result)
+			}
+		})
+	}
+}
+
+func TestIsValidService(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    sharedtypes.ServiceId
+		expected bool
+	}{
+		{
+			name:     "valid serviceId and empty serviceName",
+			input:    sharedtypes.ServiceId{Id: "Hello-1", Name: ""},
+			expected: true,
+		},
+		{
+			name:     "valid serviceId and valid serviceName",
+			input:    sharedtypes.ServiceId{Id: "SvcId", Name: "Valid Service Name"},
+			expected: true,
+		},
+		{
+			name:     "invalid serviceId and valid serviceName",
+			input:    sharedtypes.ServiceId{Id: "SvcId@", Name: "Valid Service Name"},
+			expected: false,
+		},
+		{
+			name:     "valid serviceId and invalid serviceName",
+			input:    sharedtypes.ServiceId{Id: "SvcId", Name: "Invalid Service Name@"},
+			expected: false,
+		},
+		{
+			name:     "empty serviceId and valid serviceName",
+			input:    sharedtypes.ServiceId{Id: "", Name: "Valid Name_1"},
+			expected: false,
+		},
+		{
+			name:     "valid serviceId and empty serviceName",
+			input:    sharedtypes.ServiceId{Id: "svcId", Name: ""},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := IsValidService(&tt.input)
+			if got != tt.expected {
+				t.Errorf("IsValidService(%v) = %v, want %v", tt.input, got, tt.expected)
+			}
```

Comment:
Any reason not to use `require.Errorf()`?

```suggestion
			require.Equalf(t, tt.expected, got, "IsValidService(%v)",  tt.input)
```